### DISCUSSION TITLE
Full data export

### DISCUSSION THREAD

**Discussion by johanrosenson** [plausible/analytics#2830] [Anything Else]
Hello.

I notice that the current export feature only exports aggregated data.

What is the plan to allow full export of data so that statistics can be exported from Plausible and imported into other analytics tools?

_It would be great if all similar analytics services had better import/export so that it was easier to switch between services, i think this would make it easier to also switch to Plausible from other services._

The website currently makes claims about "100% ownership of data", i don't believe this is 100% truthful unless you can take all your data with you.

* Do you have any plans to allow full export of data? 
* What is the approx. time frame for such a feature?

**Comment by metmarkosaric** [comment]
thanks for reaching out @johanrosenson! 

there are no current plans to change how our export works. it already works in the same way as our dashboard and is as flexible as you need it. you can export the default view with a click or you can segment your audience as narrow as you want and export that specific segment with a click too. the export already splits all the reports in different CSV files automatically to make it easier to take it anywhere you want

we also have a stats API which allows you to export your data programmatically. and we have integrations with Data Studio and other data warehouses so you can take your data where you want programmatically too

we don't sell the data we collect, we don't share it with anyone, we don't mine it for any behavioral insights and we claim no right to it whatsoever

**Reply by johanrosenson** [reply]
But you have all the visits in your database (ClickHouse?) right?

So technically it could (maybe) happen sometime in the future (although you have no plans for it)?

**Reply by johanrosenson** [reply]
> is as flexible as you need it

Allow me to disagree.

I have taken a proper look at the exports now, i chose a day earlier this week and selected minute precision before hitting the export button.

The files just doesn't contain enough precision to use this data in another system.

A quick example:

* I can see number of visits per minute.
* I can see how many visits individual pages got during that day.
* I cannot see when a specific page got any visits during that day.

I don't really feel like i own our data at the moment, because i can't take it with me :)

**Reply by metmarkosaric** [reply]
thanks for the feedback! as mentioned in my previous reply, our export works the same way as the dashboard itself. so if you want to see per minute visitors for a specific page, you would need to filter the dashboard by that specific page, choose minute in the interval and then click on the export icon. the stats you're looking for would be in the visitors.csv file

**Reply by johanrosenson** [reply]
Is it possible for you to respond to my first question?

> You have all the visits in your database (ClickHouse?) right?

**Reply by metmarkosaric** [reply]
yes, we use the ClickHouse database to collect the data

**Reply by johanrosenson** [reply]
So just to be clear and to finish this conversation for now; If you don't allow the customer to bring their data with them (data that you confirm that you have), that means that you (Plausible) actually own the data, not the customer. The fact that you (Plausible) are not selling/sharing the data doesn't change who actually owns it.

I will probably have to renew and keep using plausible for the foreseeable future instead of moving to a new service, because the option is to loose historical data.

Thank you for time and answers.

**Reply by metmarkosaric** [reply]
our export already works the same way as the dashboard works and you can export all your stats with it. there's also the stats api and integrations with different datewarehouses
